webfinger.go

```diff
@@ -80,3 +84,46 @@ func (wfr wfResolver) DummyUser(username string, hostname string, r []webfinger.
 func (wfr wfResolver) IsNotFoundError(err error) bool {
 	return err == wfUserNotFoundErr
 }
+
+// RemoteLookup looks up a user by handle at a remote server
+// and returns the actor URL
+func RemoteLookup(handle string) string {
+	handle = strings.TrimLeft(handle, "@")
+	// let's take the server part of the handle
+	parts := strings.Split(handle, "@")
+	resp, err := http.Get("https://" + parts[1] + "/.well-known/webfinger?resource=acct:" + handle)
+	if err != nil {
+		log.Error("Error performing webfinger request", err)
+		return ""
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Error("Error reading webfinger response", err)
+		return ""
+	}
+
+	var result map[string]interface{}
+	json.Unmarshal(body, &result)
```

Comment:
Using `interface{}` with all the forced typecasting below provides a lot of opportunity for the application to crash. Instead, we should `Unmarshal()` into a [`webfinger.Resource`](https://godoc.org/github.com/writeas/go-webfinger#Resource) struct. (See our use of this library in webfinger.go.)

Reply:
Done. The code looks way nicer now too.

If you are unable to visit the post I mentioned you in it's because it's the development writefreely instance running on my laptop and it isn't always online.